### PR TITLE
Generate canvas labels from AV logical struct DIV, if present

### PR DESCRIPTION
**Pull request by tomcrane**
Canvases usually take labels from METS physical structure.
This checks to see if there is logical structure, and if so, attempts to generate labels from that.

It will fall back to physical structure label as normal if no label can be deduced.

### CODE REVIEW

File: src/Wellcome.Dds/Wellcome.Dds.Repositories/Presentation/IIIFBuilderParts.cs [hunk 1]
```diff
@@ -489,6 +495,52 @@ public void Canvases(Manifest manifest, IManifestation metsManifestation, State
...
 
+        private static string? GetBetterAVCanvasLabel(IManifestation metsManifestation, IPhysicalFile physicalFile)
+        {
+            string? betterCanvasLabel = null;
+            // Find a better Canvas label for an AV canvas by seeing if this file has one
+            // single corresponding logical struct div. This is very conservative in deciding
+            // what a "structural" logical struct div is. Later, we might want TOCs etc - but they 
+            // can come from normal range-building behaviour.
+            var logicalStructs = metsManifestation.RootStructRange?.Children;
+            if (logicalStructs != null && logicalStructs.HasItems())
```
Comment (donaldgray) [inline src/Wellcome.Dds/Wellcome.Dds.Repositories/Presentation/IIIFBuilderParts.cs:506]: `HasItems()` will handle logicalStructs being `null`

**Review by donaldgray — APPROVED** [review]

File: src/Wellcome.Dds/Wellcome.Dds.Repositories/Presentation/IIIFBuilderParts.cs (hunk 1 above)
Comment (tomcrane) [reply src/Wellcome.Dds/Wellcome.Dds.Repositories/Presentation/IIIFBuilderParts.cs:506]: yeah, might have been an absent-minded listening to a Rider suggestion, which doesn't seem to work out that it is null checking.

Comment (tomcrane) [reply src/Wellcome.Dds/Wellcome.Dds.Repositories/Presentation/IIIFBuilderParts.cs:506]: Actually the next line requires that it isn't null

Comment (donaldgray) [reply src/Wellcome.Dds/Wellcome.Dds.Repositories/Presentation/IIIFBuilderParts.cs:506]: That should be okay though, it won't enter the if statement if it is null as `HasItems` will be false

Comment (tomcrane) [reply src/Wellcome.Dds/Wellcome.Dds.Repositories/Presentation/IIIFBuilderParts.cs]: doh..

**Pull request merged**